runtime/lua/vim/lsp.lua

```diff
@@ -0,0 +1,791 @@
+local builtin_default_server_callbacks = require 'vim.lsp.builtin_callbacks'
```

Comment:
Unless I'm misunderstanding you, that's not exactly true since the `default_server_callbacks` passed in will be `tbl_extend("keep", default_server_callbacks, builtin_server_callbacks)`, which means that we only provide ones the user doesn't provide.

I do agree that they should be available, but I thought that they would be via the require. You mean to expose them on the `lsp` object itself, right?

Reply:
I'm sorry, I overlooked that. 
Regarding the public method, I think that this is related to the future policy of the Lua modules of the Neovim project, but I think that it is better to make public APIs available without requiring them. 
This makes it easier for users to distinguish between public and private APIs.  Also, in this example, I thought it would be redundant to call 'require (' vim.lsp.builtin_callbacks ')' for something that can be called 'vim.lsp.builtin_callback'.

@bfredl What do uou think about this?